File: examples/02_gates_tour/gates_tour.cpp
// Example 02: Gates Tour — Single-qubit, multi-qubit, and rotation gates.

#include "circuit/quantumcircuit.hpp"
#include <cmath>

using namespace Qiskit::circuit;

int main() {
    QuantumRegister qr(4);
    ClassicalRegister cr(4);
    QuantumCircuit circ(qr, cr);

    // Single-qubit gates
    circ.h(0);
    circ.x(1);
    circ.y(2);
    circ.z(3);
    circ.s(0);
    circ.t(1);

    // Rotation gates
    circ.rx(M_PI / 4, 0);
    circ.ry(M_PI / 3, 1);
    circ.rz(M_PI / 2, 2);

    // Multi-qubit gates
    circ.cx(0, 1);
    circ.cz(2, 3);
    circ.swap(0, 2);
    circ.ccx(0, 1, 2);

    circ.barrier({0, 1, 2, 3});
    circ.measure(qr, cr);

    circ.print();

    return 0;
}

File: examples/02_gates_tour/gates_tour.py
"""Example 02: Gates Tour — Single-qubit, multi-qubit, and rotation gates."""

from qiskit import QuantumCircuit
import math

qc = QuantumCircuit(4, 4)

# Single-qubit gates
qc.h(0)
qc.x(1)
qc.y(2)
qc.z(3)
qc.s(0)
qc.t(1)

# Rotation gates
qc.rx(math.pi / 4, 0)
qc.ry(math.pi / 3, 1)
qc.rz(math.pi / 2, 2)

# Multi-qubit gates
qc.cx(0, 1)
qc.cz(2, 3)
qc.swap(0, 2)
qc.ccx(0, 1, 2)

qc.barrier()
qc.measure([0, 1, 2, 3], [0, 1, 2, 3])

print(qc)
